src/Microsoft.Data/DataFrame.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrame 
```

Comment:
It would be nice to write what such a thing is and what it does, and any interesting comments about its behavior/perf characteristics. I tend to do that realtively early as it helps me think...